public/app/percona/shared/core/reducers/nodes/nodes.utils.ts

```diff
@@ -54,3 +54,25 @@ export const nodeFromDbMapper = (nodeFromDb: NodeDB[]) => {
     } as Node;
   });
 };
+
+export const nodesOptionsMapper = (nodeFromDb: NodeDB[]) => {
+  return  nodeFromDb.map((node) => {
+
+    const agents = node.agents?.map(
+      (agent) => {
+        if (agent.agent_type === AgentType.pmmAgent) {
```

Comment:
I think it would be cleaner to first filter, than map, rather then the reverse:

```ts
const agents = (node.agents || [])
  .filter(agent => agent.type === AgentType.pmmAgent)
  .map<AgentOption>(agent => ({ value: agent.agent_type, label: agent.agent_id }))
```
ADD_1: You can specify the resulting type that you want from map as the first generic parameter (`map<RESULTING_TYPE>(...)`) rather than using `as`.

ADD_2: Also, setting the agents to an empty array if they are falsy (undefined/null), can lead to cleaner usage without additional check for null/undefined (as in `selectNodeAndAgent`).

Reply:
thanks for the idea